### PR TITLE
Force user to select chain on first run

### PR DESCRIPTION
**Pull request by svyatonik**
So from now users without database or with empty database or with old version of database will receive message => "You must select fork on first run: --segwit, --segwit2x, --bitcoin-cash".
After selecting fork, it can't be changed anymore. Trying to switch to another fork will lead to "Cannot select 'segwit2x' fork with non-empty database of 'segwit' fork"-like error.
After first run, fork is read from database => passing flag is not required anymore.

Example:
```
svyatonik@xwnotebook:~/dev/parity-bitcoin.test$ rm -rf db.test
svyatonik@xwnotebook:~/dev/parity-bitcoin.test$ ./pbtc -d db.test
You must select fork on first run: --segwit, --segwit2x, --bitcoin-cash
svyatonik@xwnotebook:~/dev/parity-bitcoin.test$ ./pbtc -d db.test --segwit
2017-09-01 17:13:14  INFO sync Processed 1000 blocks in 1.09 seconds (915.16 blk/s).	Peers: 6 (act: 6, idl: 0, bad: 1).	Chain: [sch:4336 -> req:640 -> vfy:24 -> stored: 1001]
2017-09-01 17:13:16  INFO sync Processed 1000 blocks in 1.19 seconds (841.61 blk/s).	Peers: 6 (act: 5, idl: 1, bad: 1).	Chain: [sch:5440 -> req:559 -> vfy:1 -> stored: 2001]
2017-09-01 17:13:17  INFO sync Processed 1000 blocks in 1.38 seconds (722.58 blk/s).	Peers: 6 (act: 6, idl: 0, bad: 0).	Chain: [sch:4160 -> req:815 -> vfy:25 -> stored: 3001]
2017-09-01 17:13:23  INFO sync Processed 1000 blocks in 5.93 seconds (168.67 blk/s).	Peers: 6 (act: 1, idl: 5, bad: 0).	Chain: [sch:5136 -> req:0 -> vfy:864 -> stored: 4001]
2017-09-01 17:13:23  INFO sync Processed 1000 blocks in 0.57 seconds (1747.58 blk/s).	Peers: 7 (act: 2, idl: 5, bad: 1).	Chain: [sch:4368 -> req:256 -> vfy:376 -> stored: 5001]
2017-09-01 17:13:25  INFO sync Processed 1000 blocks in 1.19 seconds (838.31 blk/s).	Peers: 7 (act: 7, idl: 0, bad: 0).	Chain: [sch:4832 -> req:893 -> vfy:275 -> stored: 6001]
2017-09-01 17:13:26  INFO sync Processed 1000 blocks in 1.14 seconds (879.93 blk/s).	Peers: 7 (act: 7, idl: 0, bad: 0).	Chain: [sch:4064 -> req:875 -> vfy:61 -> stored: 7001]
^C
svyatonik@xwnotebook:~/dev/parity-bitcoin.test$ ./pbtc -d db.test --segwit2x
Cannot select 'segwit2x' fork with non-empty database of 'segwit' fork
svyatonik@xwnotebook:~/dev/parity-bitcoin.test$ ./pbtc -d db.test
2017-09-01 17:13:46  WARN sync Failed to get requested block from peer#2 in 9.48 seconds.
2017-09-01 17:13:54  INFO sync Processed 1000 blocks in 16.81 seconds (59.50 blk/s).	Peers: 9 (act: 9, idl: 0, bad: 2).	Chain: [sch:3951 -> req:896 -> vfy:153 -> stored: 8074]
2017-09-01 17:13:56  WARN sync Failed to get requested block from peer#2 in 9.82 seconds.
2017-09-01 17:13:57  INFO sync Processed 1000 blocks in 2.98 seconds (336.05 blk/s).	Peers: 9 (act: 9, idl: 0, bad: 1).	Chain: [sch:4927 -> req:1024 -> vfy:49 -> stored: 9074]
2017-09-01 17:14:06  WARN sync Failed to get requested block from peer#10 in 9.73 seconds.
2017-09-01 17:14:06  WARN sync Failed to get requested block from peer#2 in 9.73 seconds.
2017-09-01 17:14:06  WARN sync Too many header failures for peer#2. Excluding from synchronization.
2017-09-01 17:14:06  WARN sync Disconnecting from peer#2 due to misbehavior: Too many header failures.
2017-09-01 17:14:07  INFO sync Processed 1000 blocks in 10.08 seconds (99.25 blk/s).	Peers: 8 (act: 0, idl: 7, bad: 1).	Chain: [sch:4415 -> req:0 -> vfy:585 -> stored: 10074]
2017-09-01 17:14:07  INFO sync Processed 1000 blocks in 0.44 seconds (2271.73 blk/s).	Peers: 8 (act: 8, idl: 0, bad: 1).	Chain: [sch:3135 -> req:733 -> vfy:132 -> stored: 11074]
```

### CODE REVIEW

File: pbtc/config.rs [hunk 1]
```diff
@@ -159,11 +164,46 @@ pub fn parse(matches: &clap::ArgMatches) -> Result<Config, String> {
...
 			verification_edge: verification_edge,
 		},
+		db: db,
 	};
 
 	Ok(config)
 }
 
+fn parse_consensus_fork(db: &db::SharedStore, matches: &clap::ArgMatches) -> Result<ConsensusFork, String> {
+	let old_consensus_fork = match db.consensus_fork() {
```
Comment (svyatonik) [inline pbtc/config.rs]: Anyone can explain to me - why
```
let old_consensus_fork = db.consensus_fork().map_err(Into::into)?;
```
doesn't work here and below? I've broke mine head trying to understand - what's wrong with this,

Comment (NikVolf) [reply pbtc/config.rs]: why not just `db.consensus_fork()?`

**Comment by NikVolf** [comment]
You've broken regtests btw, https://travis-ci.org/paritytech/parity-bitcoin/builds/270845102?utm_source=github_status&utm_medium=notification

File: pbtc/config.rs (hunk 1 above)
Comment (tomusdrw) [reply pbtc/config.rs]: You cannot have double `into` (one explicit and one from `?`)

Comment (svyatonik) [reply pbtc/config.rs]: Thanks, it is clear now :)

**Comment by svyatonik** [comment]
Thanks, @NikVolf - I'll fix tomorrow

**Comment by svyatonik** [comment]
Also updated Readme.md @5chdn - could you please check?

@5chdn @NikVolf @debris - what do you guys think about changing this:
https://github.com/paritytech/parity-bitcoin/blob/master/docker/ubuntu/Dockerfile#L36
https://github.com/paritytech/parity-bitcoin/blob/master/snap/snapcraft.yaml#L12
So following this PR logics, we can't force user to choose fork => we shouldn't hardcode any flags in these files. But without it, default commands like `docker run` will fail. I'm not sure - either it is ok to leave as is, or we should provide some default value (or maybe even 3 different dockerfiles/snaps)?

**Comment by 5chdn** [comment]
You can just append flags to `docker run`, like 

    docker run parity/parity-bitcoin --segwit

Needs some docs maybe. We should start moving some stuff from the README to the wiki anyways at some point.

**Comment by svyatonik** [comment]
So - it is ok to leave as is. Ok then, cool

**Pull request merged**